projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.html

```diff
@@ -102,7 +124,7 @@
             [igxForContainerSize]='calcHeight' [igxForItemSize]="renderedRowHeight" [igxForTrackBy]='trackChanges'
             #verticalScrollContainer (onChunkPreload)="dataLoading($event)">
             <ng-template #hierarchical_record_template>
-                <igx-hierarchical-grid-row [gridID]="id" [index]="rowIndex" [rowData]="rowData" #row>
+                <igx-hierarchical-grid-row [gridID]="id" [index]="getRowIndex(rowIndex, false)" [ghostRow]="isGhostRecord(rowData)" [rowData]="!isGhostRecord(rowData) ? rowData : rowData.recordData" #row>
```

Comment:
Is there need to calculate the index again here ( [index]="getRowIndex(rowIndex, false)" ) if it is already populated correctly in the context passed to the row ( in getContext for the template outlet)? This does not seem to be necessary for the base grid.

Reply:
Yep, it seems that when the context is passed to the igxTemplateOutlet, the index is not applied. It seem to be working with the ngTemplateOutlet, so it could be a bug in our template. As you can see in the grid the index is actually set:

https://github.com/IgniteUI/igniteui-angular/blob/master/projects/igniteui-angular/src/lib/grids/grid/grid.component.html#L161